### ISSUE TITLE
Coding conventions

### ISSUE THREAD

**Issue by sirinsidiator** [sirinsidiator/ESO-LibAddonMenu#13]
Just noticed that I used if with parentheses when the rest of the code actually does not use them and thought it might be good to write some stuff down and put it into a wiki page.

First two points:
* tab for indentation
* no parentheses on if statements

Any other ideas?

**Comment by merlight** [comment]
Agreed: no [superfluous parentheses](http://c2.com/cgi/wiki?LostInaSeaofParentheses) ;)

Pretty please:
* spaces around operators and after commas in argument lists:
```lua
local a, b = 1 + foo(3, "apples")
```

**Comment by haggen** [comment]
My suggestion is to follow the conventions ZOS used. There are some variants in there, but generally speaking it all looks pretty consistent.

- `CamelCase` for public functions and methods, `camelCase` for local variables.
- `SCREAMING_SNAKE_CASE` for constants.
- Parentheses only when needed for semantics or readability.
- Avoid one-liners, i.e. break the line after `then`, `do`, `function()`, etc.
- Space around operators.

Also, [Tabs are Evil](http://opensourcehacker.com/2012/05/13/never-use-hard-tabs/), but they use 4-space tabs.

**Comment by sirinsidiator** [comment]
Thanks to your link I moved away from tabs.
